python/sglang/srt/managers/tokenizer_manager.py

```diff
@@ -554,18 +554,43 @@ async def update_weights(
             obj.load_format = self.server_args.load_format
 
         if not self.model_update_lock.locked():
-            async with self.model_update_lock:
-                # wait for the previous generation requests to finish
-                while len(self.rid_to_state) > 0:
-                    await asyncio.sleep(0.001)
-                self.send_to_scheduler.send_pyobj(obj)
-                self.model_update_result = asyncio.Future()
-                result = await self.model_update_result
-                if result.success:
-                    self.server_args.model_path = obj.model_path
-                    self.server_args.load_format = obj.load_format
-                    self.model_path = obj.model_path
-            return result.success, result.message
+        
+            if self.server_args.dp_size == 1:
```

Comment:
@ByronHsu Any chance to reduce the code redundancy here?
The two branches seems to share many logic and code

Reply:
Yes I can do that. I did this because I want to ensure the original branch could work as before.